### DISCUSSION TITLE
Can image files uploaded to superbase storage be a vulnerability to users?

### DISCUSSION THREAD

**Discussion by branlok** [supabase/supabase#3655] [Questions] [answered]
For a simple social web app with that lets users upload images to a post to be shared publically. What are the precautions one should to prevent malicious files being uploaded? I was wondering if supabase maintain this security or is it up to us. if anyone can kindly  suggest the direction I should take? And another thing, are there any possible extensions that also brings with detecting inappropriate images?

**Comment by awalias** [comment, marked as answer]
Hi @branlok 

it is the responsibility of the app developer to comply with our [Terms of Service](https://supabase.io/docs/company/terms#3-use-restrictions). I can't advise on your specific situation (because of the complexity of jurisdiction amongst other things), but it's typical for app developers to come up with their own Terms of Service, and as you say, have some process in place to ensure that people are playing by the rules. Hope this is useful!
